### ISSUE TITLE
Smart class parameters add matcher always overrides existing matchers

### ISSUE THREAD

**Issue by mannyci** [theforeman/foreman-ansible-modules#1355]
##### SUMMARY
Puppet smart class params always overrides existing params for a smart class parameter. It removes all previous matchers and replaces with the new one

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION
```
ansible [core 2.12.1]
```

##### COLLECTION VERSION

```
theforeman.foreman 3.1.0
```

##### KATELLO/FOREMAN VERSION
```
3.18.15.0
```

##### STEPS TO REPRODUCE
```yaml
  - name: Add Smart Class parameter matcher
    theforeman.foreman.smart_class_parameter:
      server_url: "{{ satellite_server_url | default(omit) }}"
      username: "{{ satellite_username | default(omit) }}"
      password: "{{ satellite_password | default(omit) }}"
      validate_certs: "{{ satellite_validate_certs | default(omit) }}"
      puppetclass_name: "puppet_class"
      parameter: "sc_param_name"
      override_values:
        - value: "bla"
          match: "fqdn=host_fqdn.com"
```

##### EXPECTED RESULTS
Add this matcher to the sc-param matchers list

##### ACTUAL RESULTS
This matcher replaces all existing matchers and replaces only this one

**Comment by ofigenov** [comment]
I got a similar problem. Is everyone really happy with the current behavior of this module? We already have a large number of parameters configured through the Foreman interface. It is necessary to add parameters via ansible only, and not to erase everything that was configured before.

**Comment by evgeni** [comment]
This has been fixed in https://github.com/theforeman/foreman-ansible-modules/pull/1659 by providing a dedicated module to manage individual overrides.

It will be in the next release which should happen soon.

**Comment by brainptr12** [comment]
Is there any news about this bug yet?

**Comment by evgeni** [comment]
It's been released since [v3.14.0](https://github.com/theforeman/foreman-ansible-modules/releases/tag/v3.14.0), so I guess we can call it fixed?
